### PR TITLE
fixing jotai persistent state

### LINKED ISSUE #890: infinite-Loop in notification status
Details : https://www.loom.com/share/521d81b4bd0b4da5a0b62475454f4135

### PR DESCRIPTION
**Pull request by ikemHood**
# Pull Request type

Please add the labels corresponding to the type of changes your PR introduces:

- Bugfix

Resolves: #890

## Summary by CodeRabbit

- **Bug Fixes**
	- Improved the notification state update mechanism for better React state detection.

### CODE REVIEW

**Comment by ikemHood** [comment]
Hey @fricoben, how can i possibly test this fix?

I have tried performing some quest to see if i can claim a reward and perform a transaction, but the events i tried did not allow me claim rewards immediately

**Comment by ikemHood** [comment]
The previos implementation updates the array but not the reference of the array, which doesn't trigger jotai to update the store.

**Comment by ikemHood** [comment]
Hey @revengexd, Please review.

File: hooks/useNotificationManager.ts [hunk 1]
```diff
@@ -35,7 +35,7 @@ export function useNotificationManager() {
       return notification;
     });
 
-    setNotifications(updatedNotifications);
+    setNotifications([...updatedNotifications]);
```
[the remark this answers is not in the document]
Comment (ikemHood) [reply hooks/useNotificationManager.ts:38]: AI for code review  😂 😂

**Review by Marchand-Nicolas — APPROVED** [review]
Lgtm!

**Pull request merged**